frontend/src/views/DataDashboardView.vue

```diff
@@ -1,42 +1,41 @@
 <script setup lang="ts">
- import { ref, computed } from 'vue';
- import { useRoute, useRouter } from 'vue-router';
+import { ref, watch } from 'vue';
+import { useRoute, useRouter } from 'vue-router';
 
- import PresetExploreSelector from '@/components/dashboard/PresetExploreSelector.vue'
- import DDWelcomeView from '@/views/dataDashboardViews/DDWelcomeView.vue'
+import PresetExploreSelector from '@/components/dashboard/PresetExploreSelector.vue'
+import DDWelcomeView from '@/views/dataDashboardViews/DDWelcomeView.vue'
 
- import DataDashboard from '@/components/dashboard/DataDashboard.vue'
+import DataDashboard from '@/components/dashboard/DataDashboard.vue'
 
- const route = useRoute()
- const router = useRouter()
+const route = useRoute()
+const router = useRouter()
 
- const view = ref(route.params.view)
- const dataset = ref(route.params.dataset)
+const view = ref(route.params.view)
+const dataset = ref(route.params.dataset)
 
- let showWelcome = ref(false)
- if (view.value == undefined && dataset.value == undefined) {
-     showWelcome = showWelcome.value = true
- }
+const showWelcome = ref(false)
+watch([view, dataset], () => { showWelcome.value = view.value === undefined && dataset.value === undefined }, { immediate: true })
```

Comment:
we may be able to use `watchEffect` which does render on page load. See [here](https://vuejs.org/guide/essentials/watchers.html#watcheffect)

Reply:
Agreed. Same effect but cleaner code. Fixed.